textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexInfo.java

```diff
@@ -0,0 +1,82 @@
+package edu.uci.ics.textdb.dataflow.regexmatch;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @Author Zuozhi Wang
+ * @Author Shuying Lai
+ * 
+ * RegexInfo for translating regex to n-grams
+ * @see https://swtch.com/~rsc/regexp/regexp4.html for details
+ */
+public class RegexInfo {
+	boolean emptyable;
+	List<String> exact = null;
+	List<String> prefix = null;
+	List<String> suffix = null;
+	GramBooleanQuery match = null;
+	
+	/**
+	 * This initializes RegexInfo:
+	 * emptyable to false
+	 * exact, prefix, suffix to empty arraylist
+	 * match to match ALL
+	 */
+	public RegexInfo() {
+		emptyable = false;
+		exact = new ArrayList<String>();
+		prefix = new ArrayList<String>();
+		suffix = new ArrayList<String>();
+		match = new GramBooleanQuery(GramBooleanQuery.QueryOp.ANY);
+	}
+	
+	/**
+	 * @return RegexInfo describing a regex that matches NO string
+	 * This function shouldn't be called unless something goes wrong.
```

Comment:
We can declare all these classes and function as package-only, like how RE2 did. The only necessary public function is RegexToGramQueryTranslator.translate. How the algorithm works will be a block box.